app.module.js

```diff
@@ -0,0 +1,163 @@
+function App() {
+  this.thought = null
+  this.view = document.getElementById('view');
+  this.appEl = document.getElementById('app');
+  this.themeToggle = document.getElementById('theme-toggle')
+
+  this.tmplCache = {};
+  this.state = {
+    theme: 'light',
+    cachedPosts: undefined
+  }
+}
+
+Date.prototype.addHours= function(h){
+    var copiedDate = new Date(this.getTime());
+    copiedDate.setHours(copiedDate.getHours()+h);
+    return copiedDate;
+}
+
+App.prototype.init = function () {
+  var vm = this;
+
+  this.loadState()
+  .then(function(persitedState) {
+    if (persitedState) {
+      vm.state.theme = persitedState.theme;
+      vm.switchThemes(persitedState.theme, true)
+
+      vm.state.cachedPosts = persitedState.cachedPosts
+      vm.getThought(true)
+
+    } else {
+      vm.getThought(false);
+    }
+  });
+
+  this.themeToggle.addEventListener('click', function(event) {
+    var newTheme = (vm.state.theme === 'dark') ? 'light' : 'dark'
+    vm.switchThemes(newTheme)
+  })
+
+};
+
+App.prototype.getThought = function(cachedThoughts) {
+  var vm = this;
+
+  function getAndRender(t) {
+    var thought = t[Math.floor(Math.random() * t.length)];
+    vm.thought = {
+      post: thought.post,
+      author: 'u/' + thought.author,
+      link: 'http://reddit.com' + thought.permalink
+    }
+    vm.view.innerHTML = vm.renderView('main_tmpl', vm.thought)
+  }
+
+  if ((cachedThoughts &&
+    new Date() <= new Date(vm.state.cachedPosts.expirationCacheTime)) ||
+    !navigator.onLine
+  ) {
+    getAndRender(vm.state.cachedPosts.data)
+  } else {
+    vm.fetchData('https://www.reddit.com/r/showerthoughts/hot.json?limit=300')
+    .then(function (res) {
+      Object.assign(vm.state.cachedPosts, {
```

Comment:
`vm.state.cache` is initially set to `undefined`, therefore `Object.assign` can throw an error in some cases (when `localStorage` is cleared). Please revert this to the original method:

```js
vm.state.cachedPosts = {
  data: res,
  expirationCacheTime: new Date().addHours(1)
}
```